api/swagger.yaml

```diff
@@ -6302,6 +6315,75 @@ definitions:
     additionalProperties:
       type: "string"
 
+  ImageManifestSummary:
+    x-nullable: false
+    description: |
+      ImageManifestSummary represents a summary of an image manifest.
+    type: "object"
+    required: ["Id", "Descriptor", "Available", "ContentSize", "Kind"]
+    properties:
+      Id:
+        description: |
+          ID is the content-addressable ID of an image and is the same as the
+          digest of the image manifest.
+        type: "string"
+        example: "sha256:95869fbcf224d947ace8d61d0e931d49e31bb7fc67fffbbe9c3198c33aa8e93f"
+      Descriptor:
+        $ref: "#/definitions/OCIDescriptor"
+      Available:
+        description: Indicates whether all the children content is available.
+        type: "boolean"
+        example: true
+      ContentSize:
+        description: |
+          The size of the available distributable (possibly compressed) image content
+          in bytes.
+        type: "integer"
+        format: "int64"
+        example: 3987495
+      Kind:
+        description: |
+          The kind of the manifest.
+        type: "string"
+        example: "image"
+        enum:
+          - "image"
+          - "attestation"
+      ImageData:
+        description: |
+          The image data for the image manifest.
+          This field is only populated when Kind is "image".
+        type: "object"
+        required: ["Platform", "Containers", "Size", "UnpackedSize"]
+        properties:
+          Platform:
+            $ref: "#/definitions/OCIPlatform"
+          Containers:
+            description: |
+              The number of containers that are using this specific platform image.
+            type: "integer"
+            format: "int64"
+            example: 2
+          UnpackedSize:
+            description: |
+              The size of the unpacked and uncompressed image content (needed for
+              the image to be useable by containers) in bytes.
+            type: "integer"
+            format: "int64"
+            example: 3987495
+      AttestationData:
+        description: |
+          The image data for the attestation manifest.
+          This field is only populated when Kind is "attestation".
+        type: "object"
+        required: ["For"]
+        properties:
+          For:
+            description: |
+              The digest of the image manifest that this attestation is for.
+            type: "string"
+            example: "sha256:95869fbcf224d947ace8d61d0e931d49e31bb7fc67fffbbe9c3198c33aa8e93f"
```

Comment:
Since this is canonically just an annotation, could this use case be served in a more generic / less special code we maintain way by exposing the annotations directly?  (I guess we're probably doing that by having the `Descriptor` -- I'm assuming that'll be the original descriptor from the image index?)

Reply:
The `Descriptor` will have the original annotation descriptor from the image index, which looks something like:
```
    {
      "mediaType": "application/vnd.oci.image.manifest.v1+json",
      "digest": "sha256:8c8aa655b98fa8c9ef58e000ca32582bda02f188118bd7d7a40f91e54cc39f19",
      "size": 566,
      "annotations": {
        "vnd.docker.reference.digest": "sha256:e89224bfdf46aaec81e7c92b39a7f56ea68f7aed9c6ab083e7d4a17151f3a4e4",
        "vnd.docker.reference.type": "attestation-manifest"
      },
      "platform": {
        "architecture": "unknown",
        "os": "unknown"
      }
    }
```

So yeah, the `For` property is a bit redundant since that information is already in the descriptor's annotation, but thought it's much easier to expose the digest in the field instead of forcing the client to look for annotation.

From client perspective, that's useful, as it will allow correlation of the annotation manifest with the actual image manifest (annotations have unknown/unknown platform, so this is the only link between them).